server/src/main/java/io/whitefox/persistence/memory/InMemoryStorageManager.java

```diff
@@ -43,6 +45,31 @@ public InMemoryStorageManager(List<PShare> shares) {
         shares.stream().collect(Collectors.toMap(PShare::name, Function.identity())));
   }
 
+  @Override
+  public CompletionStage<Optional<Integer>> getTableVersion(
+      String share, String schema, String table, Optional<String> timestamp) {
+    var shareObj = shares.get(share);
+
+    if (shareObj == null) {
+      return CompletableFuture.completedFuture(Optional.empty());
+    } else {
+      var schemaObj = shareObj.schemas().get(schema);
+      if (schemaObj == null) {
+        return CompletableFuture.completedFuture(Optional.empty());
+      } else {
+        var tableObj =
+            schemaObj.tables().stream().filter(t -> (t.name().equals(table))).findFirst();
+        var res = tableObj.map(t -> {
+          var dl = DeltaLog.forTable(new Configuration(), t.location());
```

Comment:
I would move this logic to the service, since we want to re-use between different storage manager implementations